### ISSUE TITLE
How to use CMakeLists

### ISSUE THREAD

**Issue by ivangag** [dreamzor/xsde-cmake#1]
Hello,
sorry I'm new in cmake and my question may be stupid.
I just downloaded the xsde source and I set-up my cygwin enviroment to work correctly with cmake.

Where should I put the CMakeLists file in order to generate cxx-hybrid library (at least)?

I put it inside xsde-downloaded-packet\libxsde\ and i get the following error when run cmake 
**-- Configuring done
CMake Error at CMakeLists.txt:557 (add_library):
  Cannot find source file:
    c/expat/xmlparse.c**

I put it inside xsde-downloaded-packet\libxsde\xsde  i get the following error when run make (cmake works):

**/home/Ivan/xsde/xsde-3.2.0-i686-windows/libxsde/xsde/c/expat/xmlparse.c:5:24: fatal error: xsde/c/pre.h: No such file or directory
 #include <xsde/c/pre.h>
                        ^
compilation terminated.**
Thanks in advance for the help.

**[EDIT]**

Solved!
I had to create a directory xsde inside xsde and put in it both c and cxx folders
(/libxsde/xsde/xsde/c.. and /libxsde/xsde/xsde/cxx)
Then I had to modifiy by inserting the following line 
include_directories(".").

Maybe that tricky is not the best approach but it was the only way I found.

Ivan

**Comment by dreamzor** [comment]
Hello, @ivangag! Thanks for pointing out that the current project state is not very user-friendly. 

I've updated the project structure and downloaded library code from CodeSynthesis so it works 'out of the box' - try it out! Root README.md has also been modified.

I've also added/modified some CMakeLists.txt files so they demonstrate how it would look like in the real project.

If you have any other questions I'd be happy to help.
